client/starwhale/api/_impl/data_store.py

```diff
@@ -538,26 +623,89 @@ def delete(self, keys: List[Any]) -> None:
                     self.nbytes -= _get_size(r)
 
     def dump(self, root_path: str) -> None:
+        with self.lock:
+            schema = self.schema.copy()
         path = _get_table_path(root_path, self.table_name)
-        ensure_dir(path)
-
-        max_index = -1
-        for file in os.listdir(path):
-            type, index = _parse_parquet_name(file)
-            if type != "" and index > max_index:
-                max_index = index
-        if max_index < 0:
-            filename = "base-0.parquet"
-        else:
-            filename = f"base-{max_index + 1}.parquet"
-        _write_parquet_file(
-            os.path.join(path, filename),
-            _records_to_table(
-                self.schema,
-                list(self.records.values()),
-                list(self.deletes),
-            ),
-        )
+        if not os.path.exists(path):
```

Comment:
```
if not os.path.exists(path):
            os.mkdir(path)
```
Use  `ensure_dir`? 

1. ensure_dir will check dir existed.
2. ensure_dir will ensure the path's parent dir exists.

Reply:
Done